Dockerfile

```diff
@@ -1,23 +1,32 @@
 # Base image
-FROM nvidia/cuda:11.8.0-cudnn8-runtime-ubuntu20.04
+FROM nvidia/cuda:11.8.0-cudnn8-runtime-ubuntu22.04
+
+ENV DEBIAN_FRONTEND noninteractive
+
+# Install system dependencies
+RUN apt-get update && \
+    apt-get install -y python3-pip libpq-dev libgdal-dev gdal-bin
+
+# Verify GDAL version
+RUN if [ $(gdal-config --version | cut -d '.' -f1,2) \< "3.1" ]; then echo $(gdal-config --version) && exit 1; fi
+
+# Install Poetry
+RUN pip3 install poetry
 
 # Set working directory
 WORKDIR /app
 
-# Install dependencies
-RUN apt-get update && apt-get install -y \
-    python3 \
-    python3-pip \
-    libgl1-mesa-glx
-
-# Copy files to working directory
-COPY . .
+# Copy only pyproject.toml and poetry.lock (if exists) to cache dependencies
+COPY pyproject.toml poetry.lock* /app/
```

Comment:
`poetry.lock` must always exist!

Reply:
Generated poetry.lock file based on current dependencies.

Reply:
`poetry.lock` should be committed to git